woocommerce/woocommerce-gutenberg-products-block
# Product Gallery: Migrate placeholder to Woo default placeholder

## What

This PR migrates the custom placeholder to the Woo core placeholder.

This PR is a part of a fix to woocommerce/woocommerce#42041

## Why

Based on the exploration in woocommerce/woocommerce#42041 I have determined that the placeholder for the Product Gallery block should be updated to the core WooCommerce placeholder.

The placeholder should also support the `Crop` Product Gallery setting.

## Placeholder flow

a) When a user activates WooCommerce, it creates an attachment placeholder based on the `placeholder-attachment.png` file (1200x1200). It sets the ID as the placeholder setting in WooCommerce settings. This effectively acts as a custom placeholder.
b) When a user uploads to the Media an image and then inputs that ID manually in WooCommerce settings this acts the same way as the above scenario.
c) When a user, removes the ID completely, this is where the tricky part begins. The `wc_placeholder_img()` loads the default `placeholder.png` file (400x400) (as the placeholder fallback).
d) When a user deactivates and reactivates WooCommerce it performs scenario `a` again.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._
1. With a block theme, open `Appearance > Editor`.
2. Open `Templates > Single Product`.
3. Add the `Product Gallery` block.
5. Open a product page on the frontend with multiple product images set in the product gallery, but **no main Product Image set**. Confirm that the Product Gallery works as expected (Thumbnails, Next Previous Buttons and Pager) and that the first image is correctly set to the placeholder image.
6. Test all of the scenarios listed above the **Testing Instructions**, ensure the placeholder image displays correctly.
7. Test the `Crop` setting with a custom placeholder. I'd recommend using a vertical image as the placeholder to test.



## Screenshots or screencast



| Before | After |
| ------ | ----- |
|<img width="1096" alt="before_placeholder" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/b05b0785-45e2-4e2c-908f-187d3c91262a">|<img width="1085" alt="after_placeholder" src="https://github.com/woocommerce/woocommerce-blocks/assets/905781/84d03368-3dcf-4ed3-a5a8-d5e9a79a8ba5">|

## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Add suggested changelog entry here.
